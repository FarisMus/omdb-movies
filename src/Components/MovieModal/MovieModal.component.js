import React from 'react';
import { Image, Modal } from 'antd';

const MovieModal = (props) => {
  const { item, show, onClose } = props;

  return (
    <Modal
      title={item.Title}
      visible={show}
      onOk={onClose}
      onCancel={onClose}
    >
      <div style={{width: '100%', display: 'flex', justifyContent: 'center'}}>
        <Image src={item.Poster} preview={false}/>
      </div>
    </Modal>
  )
};

export default MovieModal;
