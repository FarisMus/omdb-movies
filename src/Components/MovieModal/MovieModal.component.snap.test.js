import React from 'react';
import renderer from 'react-test-renderer';
import MovieModal from './MovieModal.component';

jest.mock('antd', () => ({
	Modal: 'ModalComponent',
	Image: 'ImageComponent'
}));

test('Render Movie Modal Component', () => {
	const props = {
		item: {
			Title: 'Batman',
			Poster: 'batman.png'
		},
		show: true,
		onClose: jest.fn()
	};
	const component = renderer.create(
		<MovieModal {...props} />,
	);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
