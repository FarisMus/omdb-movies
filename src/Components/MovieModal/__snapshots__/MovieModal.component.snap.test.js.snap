// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Modal Component 1`] = `
<ModalComponent
  onCancel={[MockFunction]}
  onOk={[MockFunction]}
  title="Batman"
  visible={true}
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ImageComponent
      preview={false}
      src="batman.png"
    />
  </div>
</ModalComponent>
`;
